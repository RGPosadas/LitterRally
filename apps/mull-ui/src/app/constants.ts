export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
