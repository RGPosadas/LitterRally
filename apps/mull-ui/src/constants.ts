import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
