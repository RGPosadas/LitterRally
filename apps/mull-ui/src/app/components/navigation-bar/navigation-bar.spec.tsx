import React from 'react';
import { render } from '@testing-library/react';

import NavigationBar from './navigation-bar';
import { Router } from 'react-router-dom';

import { createMemoryHistory } from 'history';
import { ROUTES } from '../../../constants';

import renderer from 'react-test-renderer';

describe('NavigationBar', () => {
  it('should render successfully', () => {
    const history = createMemoryHistory();
    history.push(ROUTES.HOME);

    const { baseElement } = render(
      <Router history={history}>
        <NavigationBar />
      </Router>
    );

    expect(baseElement).toBeTruthy();
  });

  it('button associated with current should be active ', () => {
    const testIds = {
      [ROUTES.HOME]: 'home-navlink',
      [ROUTES.MAP]: 'map-navlink',
      [ROUTES.CREATE_EVENT]: 'create-event-navlink',
      [ROUTES.TOOLS]: 'tools-navlink',
      [ROUTES.PROFILE]: 'profile-navlink',
    };

    const history = createMemoryHistory();
    const dom = render(
      <Router history={history}>
        <NavigationBar />
      </Router>
    );

    for (const key in ROUTES) {
      history.push(ROUTES[key]);

      const element = dom.getByTestId(testIds[ROUTES[key]]);

      expect(element.classList.contains('active')).toBeTruthy();
    }
  });

  it('should match snapshot', () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
        <Router history={history}>
          <NavigationBar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
