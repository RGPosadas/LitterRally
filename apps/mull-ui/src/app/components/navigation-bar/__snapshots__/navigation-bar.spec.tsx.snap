// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar should match snapshot 1`] = `
<div
  className="nav-container"
>
  <img
    alt="Mull logo"
    className="logo nav-element"
    src="mull-logo.png"
  />
  <a
    aria-current={null}
    data-testid="home-navlink"
    href="/home"
    onClick={[Function]}
  >
    <svg
      className="nav-button mobile-button"
    >
      HomeIcon.svg
    </svg>
  </a>
  <a
    aria-current={null}
    data-testid="map-navlink"
    href="/map"
    onClick={[Function]}
  >
    <svg
      className="nav-button mobile-button"
    >
      MapIcon.svg
    </svg>
  </a>
  <a
    aria-current={null}
    data-testid="create-event-navlink"
    href="/create-event"
    onClick={[Function]}
  >
    <svg
      className="nav-button mobile-middle-button"
    >
      NewEventIcon.svg
    </svg>
  </a>
  <a
    aria-current={null}
    data-testid="tools-navlink"
    href="/tools"
    onClick={[Function]}
  >
    <svg
      className="nav-button mobile-button"
    >
      ToolMenuIcon.svg
    </svg>
  </a>
  <a
    aria-current={null}
    data-testid="profile-navlink"
    href="/profile"
    onClick={[Function]}
  >
    <svg
      className="nav-button mobile-button"
    >
      ProfileIcon.svg
    </svg>
  </a>
</div>
`;
